File: RentalModule.Application.Contract/CarOffers/UpdateCarOffer/UpdateCarOfferRequest.cs
using RentalModule.Domain.Models;
using RentalModule.Domain.Models.Ids;

namespace RentalModule.Application.Contract.CarOffers.UpdateCarOffer;

public record UpdateCarOfferRequest
{
    public CarOfferId? Id { get; init; }
    public string Heading { get; init; } = string.Empty;
    public string ShortDescription { get; init; }= string.Empty;
    public string FeaturedImageUrl { get; init; }= string.Empty;
    public string UrlHandle { get; init; }= string.Empty;
    public string Horsepower { get; init; }= string.Empty;
    public int YearOfProduction { get; init; }
    public string EngineDetails { get; init; }= string.Empty;
    public string DriveDetails { get; init; }= string.Empty;
    public string GearboxDetails { get; init; }= string.Empty;
    public string CarDeliverylocation { get; init; }= string.Empty;
    public string CarReturnLocation { get; init; }= string.Empty;
    public DateTime PublishedDate { get; init; }
    public bool Visible { get; init; }
    public Double OneNormalDayPrice { get; init; }
    public Double OneWeekendDayPrice { get; init; }
    public Double FullWeekendPrice { get; init; }
    public Double OneWeekPrice { get; init; }
    public Double OneMonthPrice { get; init; }
    public string[]? Tags { get; init; }
    public string[]? ImageUrls { get; init; }
    //public string[]? UnavailableDates { get; init; }
}

File: RentalApi/Application.contract/car_offers/update_car_offer/update_car_offer_request.go
package contract

import models "rental-api/Domain/models/external"

type UpdateCarOfferRequest struct {
	Id                 string   `json:"-" example:"123e4567-e89b-12d3-a456-426614174000" swaggertype:"string" validate:"required"`
	Heading            string   `json:"heading" binding:"required" example:"Updated Car Offer Title" swaggertype:"string" validate:"required"`
	ShortDescription   string   `json:"shortDescription" binding:"required" example:"Updated short description" swaggertype:"string" validate:"required"`
	FeaturedImageUrl   string   `json:"featuredImageUrl" example:"https://example.com/updated-image.jpg" swaggertype:"string"`
	UrlHandle          string   `json:"urlHandle" binding:"required" example:"updated-car-offer-title" swaggertype:"string" validate:"required"`
	Horsepower         string   `json:"horsepower" example:"350" swaggertype:"string"`
	YearOfProduction   int      `json:"yearOfProduction" example:"2023" swaggertype:"integer"`
	EngineDetails      string   `json:"engineDetails" example:"3.0L Twin-Turbo" swaggertype:"string"`
	DriveDetails       string   `json:"driveDetails" example:"RWD" swaggertype:"string"`
	GearboxDetails     string   `json:"gearboxDetails" example:"8-Speed Automatic" swaggertype:"string"`
	PublishedDate      string   `json:"publishedDate" example:"2023-12-12" swaggertype:"string"`
	Visible            bool     `json:"visible" example:"true" swaggertype:"boolean"`
	OneNormalDayPrice  float64  `json:"oneNormalDayPrice" example:"120.00" swaggertype:"number"`
	OneWeekendDayPrice float64  `json:"oneWeekendDayPrice" example:"180.00" swaggertype:"number"`
	OneWeekPrice       float64  `json:"oneWeekPrice" example:"700.00" swaggertype:"number"`
	OneMonthPrice      float64  `json:"oneMonthPrice" example:"2500.00" swaggertype:"number"`
	Tags               []string `json:"tags" example:"[\"luxury\",\"sports\"]" swaggertype:"array,string"`
	models.JwtToken    `json:",inline"`
}
